File: frontend/lib/features/tracking/models/cv_inference_response.dart
import 'cv_ingredient_item.dart';
import 'cv_suggested_dish.dart';

class CvInferenceResponse {
  CvInferenceResponse({
    required this.jobId,
    required this.requestId,
    required this.apiVersion,
    required this.status,
    this.processingTimeMs,
    this.luongTinCayChung,
    this.nguyenLieuThoQuetDuoc,
    this.danhSachMonAnGoiY,
  });

  final String jobId;
  final String requestId;
  final String apiVersion;
  final String status;
  final double? processingTimeMs;
  final String? luongTinCayChung;
  final List<CvIngredientItem>? nguyenLieuThoQuetDuoc;
  final List<CvSuggestedDish>? danhSachMonAnGoiY;

  factory CvInferenceResponse.fromJson(Map<String, dynamic> json) {
    final rawRawIngredients = json['nguyen_lieu_tho_quet_duoc'] ?? json['nguyenLieuThoQuetDuoc'];
    final rawDishes = json['danh_sach_mon_an_goi_y'] ?? json['danhSachMonAnGoiY'];
    return CvInferenceResponse(
      jobId: (json['job_id'] ?? json['jobId'] ?? '').toString(),
      requestId: (json['request_id'] ?? json['requestId'] ?? '').toString(),
      apiVersion: (json['api_version'] ?? json['apiVersion'] ?? 'v1').toString(),
      status: (json['status'] ?? '').toString(),
      processingTimeMs: _number(json['processing_time_ms'] ?? json['processingTimeMs']),
      luongTinCayChung: (json['luong_tin_cay_chung'] ?? json['luongTinCayChung'])?.toString(),
      nguyenLieuThoQuetDuoc: rawRawIngredients is List
          ? rawRawIngredients
              .whereType<Map<String, dynamic>>()
              .map(CvIngredientItem.fromJson)
              .toList()
          : null,
      danhSachMonAnGoiY: rawDishes is List
          ? rawDishes
              .whereType<Map<String, dynamic>>()
              .map(CvSuggestedDish.fromJson)
              .toList()
          : null,
    );
  }
}

double? _number(dynamic value) => value is num ? value.toDouble() : double.tryParse('$value');

File: backend/MenuGreen.BusinessLogicLayer/DTOs/Responses/CvInferenceResponse.cs
using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace MenuGreen.BusinessLogicLayer.DTOs.Responses
{
    public class CvInferenceResponse
    {
        [JsonPropertyName("job_id")]
        public string JobId { get; set; } = string.Empty;

        [JsonPropertyName("request_id")]
        public string RequestId { get; set; } = string.Empty;

        [JsonPropertyName("api_version")]
        public string ApiVersion { get; set; } = "v1";

        [JsonPropertyName("status")]
        public string Status { get; set; } = string.Empty;

        [JsonPropertyName("processing_time_ms")]
        public double? ProcessingTimeMs { get; set; }

        [JsonPropertyName("luong_tin_cay_chung")]
        public string? LuongTinCayChung { get; set; }

        [JsonPropertyName("nguyen_lieu_tho_quet_duoc")]
        public List<CvIngredientItem>? NguyenLieuThoQuetDuoc { get; set; }

        [JsonPropertyName("danh_sach_mon_an_goi_y")]
        public List<CvSuggestedDish>? DanhSachMonAnGoiY { get; set; }

        [JsonPropertyName("error_code")]
        public string? ErrorCode { get; set; }

        [JsonPropertyName("error_message")]
        public string? ErrorMessage { get; set; }

        [JsonIgnore]
        public string? Message { get; set; }
    }
}
